src/eval_utils/run.ts

```diff
@@ -0,0 +1,725 @@
+                request = {
+                    ...request,
+                    traceStatus: "complete",
+                };
+            }
+
+            if ("flow" in request) {
+                if (!_.isEqual(state!.evaluatedVersion, request.flow)) {
+                    response = await originalLog(
+                        {
+                            ...request,
+                            // @ts-ignore Log under the version expected by evaluation, not
+                            // one determined by decorators. Otherwise the evaluation will stale
+                            flow: state?.evaluatedVersion,
+                            output: undefined,
+                            error: `The version of the evaluated Flow must match the version of the callable. Expected: ${JSON.stringify(state!.evaluatedVersion)}, got: ${JSON.stringify(request.flow)}`,
+                        },
+                        options,
+                    );
+                }
+            }
+
+            if ("prompt" in request) {
+                if (!_.isEqual(state!.evaluatedVersion, request.prompt)) {
+                    response = await originalLog({
+                        ...request,
+                        // @ts-ignore Log under the version expected by evaluation, not
+                        // one determined by decorators. Otherwise the evaluation will stale
+                        prompt: state?.evaluatedVersion,
+                        output: undefined,
+                        error: `The version of the evaluated Prompt must match the version of the callable. Expected: ${JSON.stringify(state!.evaluatedVersion)}, got: ${JSON.stringify(request.prompt)}`,
+                    });
+                }
+            }
+
+            if (response === undefined) {
+                // Version validation passed, make a normal request
+                response = await originalLog(request, options);
+            }
+
+            // @ts-ignore
+            uploadCallback(response.id);
+        } else {
+            // @ts-ignore
+            response = await originalLog(request, options);
+        }
+
+        return response;
+    };
+
+    client.log = _overloadedLog.bind(client);
+
+    return client;
+}
+
+export async function runEval(
+    client: HumanloopClient,
+    file: File,
+    dataset: Dataset,
+    name?: string,
+    evaluators: Evaluator[] = [],
+    workers: number = 8,
+): Promise<EvaluatorCheck[]> {
+    // Get or create the file on Humanloop
+    if (!file.path && !file.id) {
+        throw new Error("You must provide a path or id in your `file`.");
+    }
+
+    if (file.callable && "path" in file.callable) {
+        if (file.path !== file.callable.path) {
+            throw new Error(
+                "The path of the evaluated `file` must match the path of your decorated `callable`. Expected path: " +
+                    file.path +
+                    ", got: " +
+                    file.callable.path,
```

Comment:
why `+`s and not template string? (think template string would be much cleaner)

Reply:
ack changed to template